### PR TITLE
fix multicast-forward-to-SED issue

### PR DESCRIPTION
**Pull request by librasungirl**
Unicast prefix based multicast address should be excluded from the general multicast addresses in GetIndirectTransmission();
The offset of message should be reset to 0 in case that the message would be forwarded to SED later after broadcast if the destination is unicast prefix based multicast addresses.
@jwhui , would you please help to review it?

### CODE REVIEW

File: src/core/thread/mesh_forwarder.cpp [hunk 1]
```diff
@@ -368,7 +368,8 @@ Message *MeshForwarder::GetIndirectTransmission(const Child &aChild)
         mAddMeshHeader = false;
         GetMacSourceAddress(ip6Header.GetSource(), mMacSource);
 
-        if (ip6Header.GetDestination().IsLinkLocal() || ip6Header.GetDestination().IsMulticast())
+        if (ip6Header.GetDestination().IsLinkLocal() || (ip6Header.GetDestination().IsMulticast() &&
+                                                         !(ip6Header.GetDestination().IsUnicastPrefixBasedMulticast())))
```
Comment (jwhui) [inline src/core/thread/mesh_forwarder.cpp]: Good catch.  Can we remove the multicast handling here altogether?  It seems to me that Indirect Transmissions should only be sent as MAC unicast and should never use a broadcast MAC destination.  In other words, this if-statement just checks if the IPv6 destination address is a unicast link-local address.  Thoughts?

Comment (librasungirl) [inline src/core/thread/mesh_forwarder.cpp]: Agree. Thanks @jwhui . Will submit another patch.

**Comment by librasungirl** [comment]
update mLastHeard timestamp for indirect routes along with UpdateRoutes() for direct neighbor;
fix one minor bug when 'route next is neighbor, but neighbor no longer has route'.

File: src/core/thread/mesh_forwarder.cpp [hunk 2]
```diff
@@ -368,9 +368,10 @@ Message *MeshForwarder::GetIndirectTransmission(const Child &aChild)
         mAddMeshHeader = false;
         GetMacSourceAddress(ip6Header.GetSource(), mMacSource);
 
-        if (ip6Header.GetDestination().IsLinkLocal() || ip6Header.GetDestination().IsMulticast())
+        if (ip6Header.GetDestination().IsLinkLocal() && !ip6Header.GetDestination().IsLinkLocalMulticast())
```
Comment (jwhui) [inline src/core/thread/mesh_forwarder.cpp]: I don't think the `IsLinkLocalMulticast()` check is necessary.  `IsLinkLocal()` will always return false for multicast addresses.

**Comment by jwhui** [comment]
LGTM 👍
